onnx/version_converter/adapters/group_normalization_20_21.h

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) ONNX Project Contributors
+
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// Adapter for GroupNormalization in default domain from version 20 to 21
+
+#pragma once
+
+#include <memory>
+
+#include "onnx/version_converter/adapters/adapter.h"
+
+namespace ONNX_NAMESPACE {
+namespace version_conversion {
+
+class GroupNormalization_20_21 final : public Adapter {
+ public:
+  explicit GroupNormalization_20_21() : Adapter("GroupNormalization", OpSetID(20), OpSetID(21)) {}
+
+  void transform_input(std::shared_ptr<Graph> graph, Node* node, int64_t input_id) const {
+    // Perform following sequence of ops on input, effect is similar to numpy.repeat()
+
+    //   -- Shape<start=1,end=2>() -- Div(Shape_out, num_groups)
+    //  /                                             |
+    // --  Reshape(,[-1,1]) ------------- Expand(Reshape_out,[1,Div_out]) -- Reshape(Expand_out, [-1])
+
+    Symbol kShape("Shape");
+    Node* shape = graph->create(kShape);
+    shape->i_(kstart, 1);
+    shape->i_(kend, 2);
+    shape->addInput(node->inputs()[input_id]);
```

Comment:
Shouldn't the input-index used be `0` instead of `input_id`?

Reply:
Further, the Shape and Div part is common for both scale and bias. So, it could be factored out and done once (instead of repeating it for both), right?

Reply:
The reshape shapes are also common to both scale and bias, I guess.

Reply:
It should be `input_id` since this is a helper function that's already factored out to be applied to both scale and bias.

Reply:
I am confused. Based on the description elsewhere, my thinking is this:

Input model (older opset): shapes of operands are:
X (input 0): [ N, C, ...]
scale (input 1): [num_groups]
bias (input 2): [num_groups]

We want to change scale and bias to have shape [C].

Is this understanding correct? Or, wrong?

Reply:
And, I think, the code is trying to calculate the value of `C / num_groups`. The value of `C` has to come from the shape of input 0. What am I missing?

Reply:
Ah sorry I was misled by my diagram. You're right - the Shape operator should get the input from the 0th input and the Reshape operator gets the input from the 1st and 2nd inputs. I've also corrected the diagram.

I also refactored some common code for scale and bias into the transform_input() function and left the constant nodes outside to avoid duplication. Please let me know if this is what you meant.

Reply:
@gramalingam Let me know if these changes look good to you. Thanks!